src/main/org/firebirdsql/jdbc/AbstractResultSet.java

```diff
@@ -339,11 +347,13 @@ protected void closeFields() throws SQLException {
 
         SQLExceptionChainBuilder<SQLException> chain = new SQLExceptionChainBuilder<>();
         // close current fields, so that resources are freed.
-        for (FBField field : fields) {
-            try {
-                field.close();
-            } catch (SQLException ex) {
-                chain.append(ex);
+        if (closeableFields.size() > 0) {
```

Comment:
Consider replacing this with an `if (closeableFields.isEmpty()) return;` immediately after `wasNullValid = false;`, or extend the scope of this `if` to include the `SQLExceptionChainBuilder` (if there aren't any fields to close, this builder is not necessary).